src/Components/Signup.js

```diff
@@ -0,0 +1,130 @@
+import axios from 'axios'
+import React, { Component } from 'react'
+import { Link } from 'react-router-dom'
+
+class Signup extends Component {
+
+    constructor(props) {
+        super(props)
+    
+        this.state = {
+             isError: false,
+            message: "",
+            username: "",
+            password: "",
+            messageVisibility: false
+        }
+    }
+
+    setSignupVisibilityToggle = () => {
+        this.setState(
+            (prevState)=>{
+                return {messageVisibility: !prevState.messageVisibility}
+            }
+        )
+    }
+
+    setUsername = (username) => {
+        this.setState({ username })
+    }
+
+    setPassword = (password) => {
+        this.setState({ password })
+    }
+
+    createUser = (username, password) => {
+        return axios.post(
+            'http://localhost:3100/api/users/signup',
+            {
+                username,
+                password
+            }
+        )
+    }
+
+    onSubmit = async(e) => {
+        const { username, password } = this.state
+        e.preventDefault()
+        if(!username || !password){
+            alert("Please provide username and password")
+            return
+        }
+        try{
+            const resp = await this.createUser(username, password)
+            const { data } = resp
+            const { status, message } = data
+            if(status){
+                this.setState({
+                    message,
+                     isError: false,
+                    messageVisibility: true,
+                })
+            }
+        } catch (error) {
+            let message
+            if (error.response){
+                message = error.response.data.message
+            } else {
+                message =  "Signup Failed" 
+            }
+            this.setState({
```

Comment:
let message = "Signup Failed"
if (error.response) message = error.response.data.message
this.setState .....